### ISSUE TITLE
bug: generating uuid for postgres gives 'incorrect binary data format' error

### ISSUE THREAD

**Issue by chesedo** [getsynth/synth#270]
**Describe the bug**
Trying to generate a uuid to a postgres uuid column fails with a 'incorrect binary data format' error

**To Reproduce**
1. Schema (Postgres)
```sql
CREATE TABLE test
(
	id uuid PRIMARY KEY
)
```
2. Schema (Synth)
```json
{
  "type": "array",
  "length": 1,
  "content": {
    "type": "object",
    "id": {
      "type": "string",
      "uuid": {}
    }
  }
}
```
3. See error
```
Error: At namespace "uuid"

Caused by:
    0: Failed to insert data for collection test
    1: One or more database inserts failed: error returned from database: incorrect binary data format in bind parameter 1: incorrect binary data format in bind parameter 1
```
4. Error (in the Postgres logs)
```
2021-12-23 07:18:18.091 UTC [70] ERROR:  incorrect binary data format in bind parameter 1
2021-12-23 07:18:18.091 UTC [70] CONTEXT:  unnamed portal parameter $1
2021-12-23 07:18:18.091 UTC [70] STATEMENT:  INSERT INTO test (id) VALUES
	($1);
```

**Expected behavior**
From the [Postgres integrations](https://www.getsynth.com/docs/integrations/postgres) page I expect the uuid postgres type to successfully populate from a synth string generator.

**Environment (please complete the following information):**
 - OS: Linux
 - Version: 0.6.2

**Additional context**
A successful fix should restore the [e2e test for this type](https://github.com/getsynth/synth/pull/268/files#diff-5c7f226dd02b2a0c9d5768d724c00b4eee2bbadef264763ec45e8264e54fcfcfR13)

**Comment by MrShutCo** [comment]
Are there any updates to this or workarounds? This tool seems like exactly what I'm looking for, but not being able to generate these uuids is stopping me from being able to use it

**Comment by FixKun** [comment]
> Are there any updates to this or workarounds? This tool seems like exactly what I'm looking for, but not being able to generate these uuids is stopping me from being able to use it

You can generate the data into csvs and then import those into database using `copy` command.

**Comment by MrShutCo** [comment]
Thanks for the fast reply! 
I've managed to get a workaround doing pretty much the same thing, exporting to json and then using a JS library that can just insert the JSON into pg (https://knexjs.org/)

**Comment by ThomWright** [comment]
It seems `sqlx` converts strings to binary before sending them to the database. The binary representation PostgreSQL expects for the `uuid` type is different from the bytes representing the formatted string.

This means we can't give `sqlx` a string representation of a UUID for a `uuid` column.

See the `Encode` impl for [Uuid](https://docs.rs/sqlx-core/0.5.13/src/sqlx_core/postgres/types/uuid.rs.html#23-29) and [str](https://docs.rs/sqlx-core/0.5.13/src/sqlx_core/postgres/types/str.rs.html#76-82). Both get converted to bytes, but with different implementations.


Currently UUIDs are generated using the "string" generator.

I haven't managed to get my head around Synth's internal data model yet, but I'm wondering whether the `core::graph::Value` enum should have another variant to distinguish `Uuid`s from `String`s?

**Comment by IgorAndrejewZETO** [comment]
I have same issue while trying to generate a uuid to a postgres.

**Comment by chesedo** [comment]
@ThomWright you are correct, `core::graph::Value` should have another variant for `Uuid`s

**Comment by balamuralisrinivasan** [comment]
This tool is apt for my needs. however I am struck with the same UUID problem.  Since there are multiple parent-child relationships, i am unable to use the workaround like CSV, JSON etc. Can a fix be provided?

**Comment by justneedham** [comment]
Ditto ran into the same bug today. Bummer. If I knew Rust I'd chip in. I think Synth is such a great idea

**Comment by iamwacko** [comment]
Yeah, this has been a problem for a while. I will have more time to work on this soon, so I hope this will be fixed soon.

**Comment by CMCDragonkai** [comment]
Our database uses alot of foreign keys, without supporting uuid, there's no way to use synth to generate https://www.getsynth.com/docs/content/same-as data for different tables, since our primary keys of our tables are UUID.

**Comment by janmullerbb** [comment]
Hi, any progress on it? Our db uses only UUID as a references (ids) so now we can't use synth at all :-(.
